WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -660,14 +661,18 @@ private Script GetCoordinatorScriptPreventReuse(Round round)
 		return coordinatorScriptPubKey;
 	}
 
-	private void CoinVerifier_CoinBlacklisted(object? _, Coin coin)
+	private async void CoinVerifier_CoinBlacklistedAsync(object? _, Coin coin)
 	{
-		// For logging reason Prison needs the roundId.
-		var roundState = RoundStates.FirstOrDefault(rs => rs.CoinjoinState.Inputs.Any(input => input.Outpoint == coin.Outpoint));
+		using CancellationTokenSource cts = new(TimeSpan.FromMinutes(1));
+		using (await AsyncLock.LockAsync(cts.Token).ConfigureAwait(false))
```

Comment:
Are you sure about using this lock here? Why not taking a snapshot of rounds? `var round = Rounds.ToList().FirstOrDefault...`

Reply:
As I wrote, I was not sure. In Arena, whenever we touch the `Rounds` we use the lock, so I followed this.

If I can get away with taking a snapshot of the rounds and use that, that would simplify this PR a lot.